x-pack/libbeat/common/aws/credentials.go

```diff
@@ -53,6 +54,13 @@ func InitializeAWSConfig(config ConfigAWS) (awssdk.Config, error) {
 			AWSConfig.Region = "us-east-1"
 		}
 	}
+
+	// Assume IAM role if iam_role config parameter is given
+	if config.RoleArn != "" {
+		logger.Debug("Switching credentials provider to AssumeRoleProvider")
```

Comment:
nit: I would move the logging inside `switchToAssumeRoleProvider`, in case we will invoke the function somewhere else in the future